pkg/pod/entrypoint_lookup_impl_test.go

```diff
@@ -0,0 +1,77 @@
+package pod
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	remotetest "github.com/tektoncd/pipeline/test"
+	"github.com/tektoncd/pipeline/test/parse"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestGet(t *testing.T) {
```

Comment:
I'm always glad to have more tests, and this should ensure the code is at least exercised, but it doesn't currently really ensure that the specified imagePullSecret is used when making requests to the registry. I think it's worth keeping, but maybe we should document its limitations so we don't get a false sense of security here.

Reply:
I have been working on trying to address. Please let me know if the latest change makes sense or not. 😄 
Thanks!